src/subscription-manager.ts

```diff
@@ -0,0 +1,175 @@
+import type { Dwn, SubscriptionFilter } from '@tbd54566975/dwn-sdk-js';
+import type { EventMessage, PermissionsGrant } from '@tbd54566975/dwn-sdk-js';
+
+import type { JsonRpcSuccessResponse } from './lib/json-rpc.js';
+import { SubscriptionRequest } from '@tbd54566975/dwn-sdk-js';
+import type { SubscriptionRequestReply } from '@tbd54566975/dwn-sdk-js';
+import type WebSocket from 'ws';
+import { WebSocketServer } from 'ws';
+import { v4 as uuidv4 } from 'uuid';
+
+export class Subscription {
+  from?: string;
+  subscriptionId: string;
+  createdAt: string;
+  description: string;
+  filters?: SubscriptionFilter[];
+  permissionGrant: PermissionsGrant;
+  connection: WebSocket;
+}
+
+export interface SubscriptionController {
+  clear(): Promise<void>;
+  close(): Promise<void>;
+  start(): Promise<void>;
+  subscribe(
+    request: RegisterSubscriptionRequest,
+  ): Promise<RegisterSubscriptionReply>;
+}
+
+export type RegisterSubscriptionRequest = {
+  from: string; // from connection
+  socket: WebSocket; // socket connection
+  filters?: SubscriptionFilter[]; // filters, if applicable
+  permissionGrant?: PermissionsGrant; //permission grant, if applicable
+  request: SubscriptionRequest; // subscription request
+};
+
+export type RegisterSubscriptionReply = {
+  reply?: SubscriptionRequestReply;
+  subscriptionId?: string;
+};
+
+export type defaultSubscriptionChannel = 'event';
+
+export type SubscriptionManagerOptions = {
+  wss?: WebSocketServer;
+  dwn: Dwn;
+  tenant: string;
+};
+
+export class SubscriptionManager {
+  private wss: WebSocketServer;
+  private dwn: Dwn;
+  private connections: Map<string, Subscription>;
```

Comment:
keep track of all the subscriptions here. IMPORTANT is to close the subscription when it's finished.